### DISCUSSION TITLE
Graph generation from PDB with non-standard names for titratable residues

### DISCUSSION THREAD

**Discussion by mverissi** [a-r-j/graphein#285] [Q&A] [answered]
Hi,

I have recently installed Graphein and I would like to start using it on a problem I have in mind. 

In files downloaded from PDB resolved by NMR, one has the hydrogen atoms resolved. Therefore, it's possible to know if, for example, an `ASP` residue is neutral (Amber's `ASH`) or protonated (Amber's`ASP`). However, in a file downloaded from PDB, all of them will have `ASP` labels (which I will refer to as “standard labelling”, as opposed to `ASH`, to which I will refer to as “non-standard labelling”), regardless of their protonation state.

From the website, I see Graphein is capable of determining what kinds of interactions two different residues have (peptide backbone, H-Bonds, Disulphide, ionic…). If I'm not mistaken, this is done by calculating the distances, and perhaps angles, between atoms (please correct me if I'm wrong). 

Is it mandatory that residues have the standard labelling for this determination, or would Graphein correctly determine the interactions with non-standard labelling, as in the example I gave for `ASP`? I would suppose that the most important thing in this case would be the _atom_ labelling, and I would appreciate if you could clarify this.

Best,

Marcos Verissimo Alves

**Comment by a-r-j** [comment, marked as answer]
Hi @mverissi 

Great question! You are correct that these assignments are based on distances and angles. The answer is it depends and you'll need to look at the function definition to make that determination.

If you look at the definition of say, [H bonds](https://github.com/a-r-j/graphein/blob/82d3ba333c66f97ceeff0c630427988cf410960b/graphein/protein/edges/distance.py#L335), we can see that it simply depends on atom naming so these would be picked up.

However, if we look at [ionic interactions](https://github.com/a-r-j/graphein/blob/82d3ba333c66f97ceeff0c630427988cf410960b/graphein/protein/edges/distance.py#L375), these are determined (in part) based on [residue labelling](https://github.com/a-r-j/graphein/blob/82d3ba333c66f97ceeff0c630427988cf410960b/graphein/protein/resi_atoms.py#L1016).

However, it would actually be very straightforward to add support for e.g. `ASH`, you would simply need to add the relevant residue names to the lists in [resi_atoms.py](https://github.com/a-r-j/graphein/blob/master/graphein/protein/resi_atoms.py). 

Without making any changes to the Graphein source code (though I'd be super appreciative of a PR :grin:), you could simply do something like:

```python
import graphein.protein as gp
gp.resi_atoms.IONIC_RESIS = ["ARG", "LYS", "HIS", "ASP", "GLU", "ASH"] # N.b. we have added "ASH"

gp.construct_graph(pdb_code="4hhb", edge_construction_functions = [gp.add_ionic_interactions])
```

**Reply by mverissi** [reply]
Oh wow, seems like this would be simple indeed. Thanks so much for the prompt answer, @a-r-j !

I will make no promises because although I am developing code, I'm using the packages that are already available, and the problem I have in mind is already a side project I'm struggling to get in motion. However, if I do have the time, I will try to implement this.
